lib/perl/Genome/InstrumentData/Command/Import/Inputs.pm

```diff
@@ -0,0 +1,110 @@
+package Genome::InstrumentData::Command::Import::Inputs;
+
+use strict;
+use warnings;
+
+use Genome;
+
+use Genome::InstrumentData::Command::Import::Inputs::SourceFiles;
+
+class Genome::InstrumentData::Command::Import::Inputs {
+    is => 'UR::Object',
+    id_by => [ # use arrayref to keep order
+        process_id => { is => 'Text', },
+        line_number => { is => 'Number', },
+    ],
+    has => {
+        entity_params => {
+            is => 'HASH',
+            default_value => {
+                individual => {}, sample => {}, library => {},
+            },
+        },
+        source_paths => { is => 'Text', is_many => 1, },
+    },
+    has_optional => {
+        analysis_project_id => { is => 'Text', },
+    },
+    has_transient => {
+        format => { via => 'source_files', to => 'format', },
+        library_name => { via => 'library', to => 'name', },
+        sample_name => { via => 'library', to => 'sample_name', },
+    },
+};
+
+sub get { shift; Genome::InstrumentData::Command::Import::Inputs::Factory->get->from_inputs_id(@_); }
+sub create { Carp::confess('Use inputs factory to create!'); }
+
+sub lib_and_source_file_md5sum {
+    my $self = shift;
+    return substr(
+        Genome::Sys->md5sum_data( join(' ', $self->library_name, $self->source_paths) ), 
+        0, 6,
+    );
+}
+
+sub analysis_project {
+    my $self = shift;
+    if ( $self->process ) {
+        return $self->process->analysis_project;
+    }
+    elsif ( $self->analysis_project_id ) {
+        return Genome::Config::AnalysisProject->get(id => $self->analysis_project_id);
+    }
+    $self->fatal_message('No process or analysis_project_id to get analysis project!');
+}
+
+sub library {
+    my $self = shift;
+    my %params;
+    if ( $self->entity_params->{library}->{id} ) {
+        %params = ( id => $self->entity_params->{library}->{id} );
+    }
+    elsif ( $self->entity_params->{library}->{name} ) {
+        %params = ( name => $self->entity_params->{library}->{name} );
+    }
+    else {
+        $self->fatal_message('No library id or name given to inputs!');
+    }
+    return Genome::Library->get(%params);
+}
+
+sub process {
+    return Genome::InstrumentData::Command::Import::Process->get(id => $_[0]->process_id);
```

Comment:
This could be turned into an `id_by => 'process_id'` property in the class definition.